optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/optional/score/MeetingSchedulingConstraintProvider.java

```diff
@@ -0,0 +1,224 @@
+                        filtering((requiredAttendance, preferredAttendance, leftAssignment,
+                                rightAssignment) -> rightAssignment.calculateOverlap(leftAssignment) > 0))
+                .penalizeConfigurable("Required and preferred attendance conflict");
+    }
+
+    protected Constraint preferredAttendanceConflict(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(PreferredAttendance.class)
+                .join(PreferredAttendance.class,
+                        equal(PreferredAttendance::getPerson, PreferredAttendance::getPerson),
+                        lessThan(PreferredAttendance::getId, PreferredAttendance::getId))
+                .join(MeetingAssignment.class,
+                        filtering((leftAttendance, rightAttendance,
+                                leftAssignment) -> leftAttendance.getMeeting() == leftAssignment.getMeeting()),
+                        filtering((leftAttendance, rightAttendance,
+                                leftAssignment) -> leftAssignment.getStartingTimeGrain() != null))
+                .join(MeetingAssignment.class,
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> rightAttendance.getMeeting() == rightAssignment.getMeeting()),
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> rightAssignment.calculateOverlap(leftAssignment) > 0))
+                .penalizeConfigurable("Preferred attendance conflict");
+    }
+
+    // ************************************************************************
+    // Soft constraints
+    // ************************************************************************
+
+    protected Constraint doMeetingsAsSoonAsPossible(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(MeetingAssignment.class)
+                .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() != null)
+                .penalizeConfigurable("Do all meetings as soon as possible",
+                        MeetingAssignment::getLastTimeGrainIndex);
+    }
+
+    protected Constraint oneBreakBetweenConsecutiveMeetings(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(MeetingAssignment.class)
+                .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() != null)
+                .join(MeetingAssignment.class,
+                        filtering((leftAssignment, rightAssignment) -> rightAssignment.getStartingTimeGrain() != null),
+                        filtering((leftAssignment,
+                                rightAssignment) -> leftAssignment
+                                        .getLastTimeGrainIndex() == rightAssignment.getStartingTimeGrain().getGrainIndex() - 1))
```

Comment:
I think this should be doable via `equal()` as well.

Reply:
I don't see it, is there a way to use `Predicate`s with `equal()`?